shell/platform/linux/fl_renderer.h

```diff
@@ -14,6 +14,34 @@
 
 G_BEGIN_DECLS
 
+// Struct holding the state of an individual pointer. The engine doesn't keep
+// track of which buttons have been pressed, so it's the embedding's
+// responsibility.
+struct PointerState {
```

Comment:
This struct seems to be in the wrong place, as it relates to the engine, not the renderer. To match existing code like `fl_engine_send_mouse_pointer_event` this should probably be an internal struct and not in a header.

Reply:
I moved it to the engine, but not 100% sure what you meant.. Can you help me with this, please?

Reply:
This structure is only used in `fl_engine.cc`, therefore it should be defined in there. It's currently in a header, which suggests it's part of external API for a module.

Reply:
Yes, thanks for the clarification!